### ISSUE TITLE
Duplicated hitsplats on Tormented Demons

### ISSUE THREAD

**Issue by Brew-Time-RS** [Extended-Hitsplats/extended-hitsplats#12]
When the plugin is turned on, it shows 2 hitsplats of the same value when attacking Tormented Demons
![td hitsplat](https://github.com/user-attachments/assets/ebc4c453-6231-456b-a951-9226e6806c82)

**Comment by YeahImRose** [comment]
Hi, still trying to figure out a fix for this but will update when it's fixed!

**Comment by Grantapher** [comment]
Is there a root cause figured out? I was thinking of taking a stab at it but perhaps you already have all or most of a fix.

**Comment by YeahImRose** [comment]
> Is there a root cause figured out? I was thinking of taking a stab at it but perhaps you already have all or most of a fix.

The main problem is that there's not a direct way to hide the default hitsplats. I've tried replacing the default hitsplat sprites with blank image data but that still leads to the text of the number on the hitsplat being drawn. There's no way to interrupt that process from my investigations either.

The solution that seems best would be trying to figure out the system by which the default hitsplats are displayed. It'd have to be exact, and vanilla hitsplats behavior is strange and not well documented. If you can perfectly replicate the placement, duration, and timing of the vanilla hitsplats then it's possible to draw over top of them, but that appears to be the only way to implement a fix for this.

I've done quite a lot of tweaking of the position calculations and timings but working around the combination of RuneLite's client tick system and the game ticks makes everything get out of sync pretty easily.

More than happy to take any ideas or observations about hitsplat ordering you do find though! @Grantapher

**Comment by chrisreydev** [comment]
@YeahImRose I'm looking into your code because the hitsplats render under the hp bar I made. It looks like it might be a lost cause unless we rerender everything 2d (entity hider can hide all 2d including hitsplats numbers).

Funnily, I actually used the same method you mentioned with the hp bar... I replaced the hp sprite with a blank one because i was running into the same issue with HP Bar alignment.

the plugin if you are curious:
https://github.com/chrisreydev/advanced-hp-bar-plugin

**Comment by YeahImRose** [comment]
> [@YeahImRose](https://github.com/YeahImRose) I'm looking into your code because the hitsplats render under the hp bar I made. It looks like it might be a lost cause unless we rerender everything 2d (entity hider can hide all 2d including hitsplats numbers).
> 
> Funnily, I actually used the same method you mentioned with the hp bar... I replaced the hp sprite with a blank one because i was running into the same issue with HP Bar alignment.
> 
> the plugin if you are curious: https://github.com/chrisreydev/advanced-hp-bar-plugin

Yeah and it's an extra bummer because the actual text of the hitsplat renders separately and I couldn't find any way to hide it. If that was possible it'd be trivial to just fully re-implement the hitsplats 😢
